File: 2021/day-02/main_test.go
package main

import "testing"

func TestPartOne(t *testing.T) {
	input := []string{
		"forward 5",
		"down 5",
		"forward 8",
		"up 3",
		"down 8",
		"forward 2",
	}
	got := PartOne(input)
	want := 150
	if got != want {
		t.Errorf("got %q, wanted %q", got, want)
	}
}

func TestPartTwo(t *testing.T) {
	input := []string{
		"forward 5",
		"down 5",
		"forward 8",
		"up 3",
		"down 8",
		"forward 2",
	}
	got := PartTwo(input)
	want := 900
	if got != want {
		t.Errorf("got %q, wanted %q", got, want)
	}
}

File: 2021/day-01/main_test.py
import unittest

import main


class TestCase(unittest.TestCase):
    def setUp(self):
        self.input = [199, 200, 208, 210, 200, 207, 240, 269, 260, 263]

    def test_part_one(self):
        self.assertEqual(main.part_one(self.input), 7)

    def test_part_two(self):
        self.assertEqual(main.part_two(self.input), 5)
